server/API/ListLectures.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+$studentQuery = <<<'EOC'
+SELECT
+	lectureId,
+	courseId,
+	courseName,
+	startTS,
+	endTS,
+	online,
+	teacherName,
+	roomName,
+	IFNULL(MIN(COUNT(*), totalSeats), 0) as bookedSeats,
+	totalSeats,
+	IFNULL(SUM(isMe), 0) as bookedSelf,
+	IFNULL(NOT (bookingOrder <= totalSeats AND SUM(isMe) = 1), 0) as inWaitingList
+FROM (
+	SELECT
+		L.ID as lectureId,
+		C.ID as courseId,
+		C.name as courseName,
+		L.start_ts as startTS,
+		L.end_ts as endTS,
+		L.settings & :online as online,
+		U.lastname || ' ' || U.firstname as teacherName,
+		R.name as roomName,
+		B.user_id as bookedUser,
+		R.seats as totalSeats,
+		B.user_id = :studentId as isMe,
+		RANK() OVER (PARTITION BY L.ID ORDER BY B.booking_ts ASC) AS bookingOrder 
+	FROM
+		lectures L, rooms R, courses C, users U
+		LEFT JOIN bookings B
+		ON B.lecture_id = L.ID
+	WHERE
+		L.room_id = R.ID AND
+		L.course_id = C.ID AND
+		C.teacher_id = U.ID AND
+		B.cancellation_ts IS NULL AND
+		L.settings & :cancelled = 0 AND
+		L.start_ts >= :startTS AND
+		L.start_ts <= :endTS
+	GROUP BY L.ID, B.booking_ts
+	ORDER BY lectureID, bookingOrder
+)
+GROUP BY
+	lectureId
+EOC;
+
+$teacherQuery = <<<'EOC'
+SELECT
+	lectureId,
+	courseId,
+	courseName,
+	startTS,
+	endTS,
+	online,
+	teacherName,
+	roomName,
+	IFNULL(MIN(COUNT(*), totalSeats), 0) as bookedSeats,
+	IFNULL(MAX(0, COUNT(*) - totalSeats), 0) as waitingUsers,
+	totalSeats
+FROM (
+	SELECT
+		L.ID as lectureId,
+		C.ID as courseId,
+		C.name as courseName,
+		L.start_ts as startTS,
+		L.end_ts as endTS,
+		L.settings & :online as online,
+		U.lastname || ' ' || U.firstname as teacherName,
+		R.name as roomName,
+		B.user_id as bookedUser,
+		R.seats as totalSeats
+	FROM
+		lectures L, rooms R, courses C, users U
+		LEFT JOIN bookings B
+		ON B.lecture_id = L.ID
+	WHERE
+		L.room_id = R.ID AND
+		L.course_id = C.ID AND
+		C.teacher_id = U.ID AND
+		B.cancellation_ts IS NULL AND 
+		U.ID = :teacherId AND 
+		L.settings & :cancelled = 0 AND
+		L.start_ts >= :startTS AND
+		L.start_ts <= :endTS
+)
+GROUP BY
+	lectureId
+EOC;
+
+$bookMngrQuery = <<<'EOC'
+SELECT
+	lectureId,
+	courseId,
+	courseName,
+	startTS,
+	endTS,
+	online,
+	teacherName,
+	roomName,
+	IFNULL(MIN(SUM(isBooked), totalSeats), 0) as bookedSeats,
+	IFNULL(MAX(0, SUM(isBooked) - totalSeats), 0) as waitingUsers,
+	IFNULL(SUM(isCancelled), 0) as cancelledBookings,
+	totalSeats
+FROM (
+	SELECT
+		L.ID as lectureId,
+		C.ID as courseId,
+		C.name as courseName,
+		L.start_ts as startTS,
+		L.end_ts as endTS,
```

Comment:
shouldn't the endpoint also give the year and semester information about the course?